tests/std/tests/VSO_0000000_vector_algorithms/test.cpp

```diff
@@ -164,10 +197,18 @@ void test_case_min_max_element(const vector<T>& input) {
 
 template <class T>
 void test_min_max_element(mt19937_64& gen) {
-    using Distribution = conditional_t<is_floating_point_v<T>, uniform_real_distribution<T>,
-        conditional_t<(sizeof(T) > 1), uniform_int_distribution<T>, uniform_int_distribution<int>>>;
-
-    Distribution dis(1, 20);
+    static constexpr T Min = numeric_limits<T>::min();
+    static constexpr T Max = numeric_limits<T>::max();
+
+    auto dis = [] {
+        if constexpr (is_floating_point_v<T>) {
+            return uniform_real_distribution<T>{-Max / 2, Max / 2};

```

Comment:
This isn't a great distribution for generating random test cases. It practically generates only huge floating point numbers besides 0.

Reply:
We may drop the real distribution tests entirely.
The floating point optimization #2439 is not implemented yet anyway.

Reply:
Agreed, I'll cauterize this.